frontend/src/__tests__/cypress/cypress/utils/users.ts

```diff
@@ -0,0 +1,75 @@
+import { mockStatus } from '~/__mocks__';
+import { mockSelfSubjectAccessReview } from '~/__mocks__/mockSelfSubjectAccessReview';
+import { SelfSubjectAccessReviewModel } from '~/__tests__/cypress/cypress/utils/models';
+
+const interceptSelfSubjectAccessReview = ({
+  isClusterAdmin,
+  isProjectAdmin,
+  isSelfProvisioner,
+  projects,
+}: {
+  isClusterAdmin?: boolean;
+  isProjectAdmin?: boolean;
+  isSelfProvisioner?: boolean;
+  projects?: string[];
+} = {}) => {
+  cy.interceptK8s('POST', SelfSubjectAccessReviewModel, (req) => {
+    const { verb, group, resource, namespace } = req.body.spec.resourceAttributes;
+    req.reply(
+      mockSelfSubjectAccessReview({
+        verb,
+        group,
+        resource,
+        allowed: isClusterAdmin
+          ? true
+          : resource === 'projectrequests'
+          ? isSelfProvisioner
+          : // only project admins can create rolebindings
+          resource === 'rolebindings' &&
+            ['*', 'create', 'delete', 'deletecollection', 'update', 'patch'].includes(verb)
+          ? isProjectAdmin
+          : namespace && (!projects || projects.includes(namespace)),
+      }),
+    );
+  });
+};
+
+/**
+ * Disallowed users have no access to the dashboard.
+ */
+export const asDisallowedUser = (): void => {
+  cy.interceptOdh('GET /api/status', mockStatus({ isAllowed: false }));
+  interceptSelfSubjectAccessReview();
+};
+
+/**
+ * Cluster admins have access to perform any action on resource in every project.
+ */
+export const asClusterAdminUser = (): void => {
```

Comment:
Ah I see -- you're aiming explicitly at `cluster-admin` -- recommend adding `asProductAdmin` as well. Which should not grant access to any project but will the settings stuff. Not sure if that's possible with this current infra atm.

Reply:
I will add `asProductAdmin` as suggested.